Mrfixit: genimage is programmed to copy the entire rootfs in order to use mountpoints, but not all configs need mountpoints (rockchip does not), so only copy rootfs if needed. When not needed, skipping this uses less disk space and increases the speed an image can be created.
--- a/genimage.c	2018-09-11 12:55:06.191771954 -0400
+++ b/genimage.c	2018-09-11 12:53:39.399771000 -0400
@@ -414,38 +414,42 @@
 {
 	struct image *image;
 	struct mountpoint *mp;
-	int ret;
-
-	add_root_mountpoint();
-
-	ret = systemp(NULL, "mkdir -p \"%s\"", tmppath());
-	if (ret)
-		return ret;
-
-	ret = systemp(NULL, "cp -a \"%s\" \"%s/root\"", rootpath(), tmppath());
-	if (ret)
-		return ret;
+	int ret, mnts = 0;
 
 	list_for_each_entry(image, &images, list) {
-		if (image->mountpoint)
+		if (image->mountpoint) {
 			image->mp = add_mountpoint(image->mountpoint);
+			mnts = mnts + 1;
+		}
 	}
 
-	list_for_each_entry(mp, &mountpoints, list) {
-		if (!strlen(mp->path))
-			continue;
-		ret = systemp(NULL, "mv \"%s/root/%s\" \"%s\"", tmppath(), mp->path, mp->mountpath);
-		if (ret)
-			return ret;
-		ret = systemp(NULL, "mkdir \"%s/root/%s\"", tmppath(), mp->path);
-		if (ret)
-			return ret;
-		ret = systemp(NULL, "chmod --reference=\"%s\" \"%s/root/%s\"", mp->mountpath, tmppath(), mp->path);
+	if (mnts > 0) {
+		add_root_mountpoint();
+
+		ret = systemp(NULL, "mkdir -p \"%s\"", tmppath());
 		if (ret)
 			return ret;
-		ret = systemp(NULL, "chown --reference=\"%s\" \"%s/root/%s\"", mp->mountpath, tmppath(), mp->path);
+
+		ret = systemp(NULL, "cp -a \"%s\" \"%s/root\"", rootpath(), tmppath());
 		if (ret)
 			return ret;
+
+		list_for_each_entry(mp, &mountpoints, list) {
+			if (!strlen(mp->path))
+				continue;
+			ret = systemp(NULL, "mv \"%s/root/%s\" \"%s\"", tmppath(), mp->path, mp->mountpath);
+			if (ret)
+				return ret;
+			ret = systemp(NULL, "mkdir \"%s/root/%s\"", tmppath(), mp->path);
+			if (ret)
+				return ret;
+			ret = systemp(NULL, "chmod --reference=\"%s\" \"%s/root/%s\"", mp->mountpath, tmppath(), mp->path);
+			if (ret)
+				return ret;
+			ret = systemp(NULL, "chown --reference=\"%s\" \"%s/root/%s\"", mp->mountpath, tmppath(), mp->path);
+			if (ret)
+				return ret;
+		}
 	}
 
 	/*
